pkg/utils/test/envt/envt.go

```diff
@@ -0,0 +1,139 @@
+package envt
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlCli "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/client"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/scheme"
+	"github.com/opendatahub-io/opendatahub-operator/v2/tests/envtestutil"
+)
+
+type OptionFn func(in *EnvT)
+
+func WithScheme(value *runtime.Scheme) OptionFn {
+	return func(in *EnvT) {
+		in.s = value
+	}
+}
+
+func WithProjectRoot(value string, elem ...string) OptionFn {
+	return func(in *EnvT) {
+		r := value
+		if l := len(elem); l != 0 {
```

Comment:
Is it to save one string copy in case of one value?

Reply:
just overthinking, fixed